osgp/platform/osgp-adapter-domain-core/src/main/java/org/opensmartgridplatform/adapter/domain/core/application/services/FirmwareManagementService.java

```diff
@@ -89,13 +107,74 @@ public void updateFirmware(final CorrelationIds ids,
         this.findOrganisation(ids.getOrganisationIdentification());
         final Device device = this.findActiveDevice(ids.getDeviceIdentification());
 
+        if (device instanceof Ssld) {
+            final Ssld ssld = this.findSsldForDevice(device);
+            this.createSsldPendingFirmwareUpdateRecord(ids, ssld,
+                    firmwareUpdateMessageDataContainer.getFirmwareModuleData(),
+                    firmwareUpdateMessageDataContainer.getFirmwareUrl());
+        }
+
         this.osgpCoreRequestMessageSender.send(
                 new RequestMessage(ids,
                         this.domainCoreMapper.map(firmwareUpdateMessageDataContainer,
                                 org.opensmartgridplatform.dto.valueobjects.FirmwareUpdateMessageDataContainer.class)),
                 messageType, messagePriority, device.getIpAddress(), scheduleTime);
     }
 
+    private void createSsldPendingFirmwareUpdateRecord(final CorrelationIds ids, final Ssld ssld,
+            final FirmwareModuleData firmwareModuleData, final String firmwareUrl) {
+        try {
+            final String[] split = firmwareUrl.split("/");
+            Assert.isTrue(split.length >= 1, "Splitting URL on / failed!");
+            final String firmwareFilename = split[split.length - 1];
+
+            final List<FirmwareFile> firmwareFiles = this.firmwareFileRepository.findByFilename(firmwareFilename);
+            Assert.isTrue(firmwareFiles.size() == 1, "Expected 1 firmware file for filename: " + firmwareFilename);
+            final FirmwareFile firmwareFile = firmwareFiles.get(0);
+
+            final FirmwareFileFirmwareModule firmwareFileFirmwareModule = this.firmwareFileFirmwareModuleRepository
+                    .findByFirmwareFile(firmwareFile);
+            final String firmwareVersion = firmwareFileFirmwareModule.getModuleVersion();
+
+            final FirmwareModuleType firmwareModuleType = firmwareModuleData.getFirmwareModuleType();
+
+            SsldPendingFirmwareUpdate ssldPendingFirmwareUpdate = new SsldPendingFirmwareUpdate(true,
```

Comment:
Consider using the correlation id from the update firmware request while creating the pending firmware update.
In that case we don't need a setter for it in the class (making the object immutable).